File: Sources/UIButton/UIButton+Loading.m
//
//  UIButton+Loading.m
//  Transport
//
//  Created by LSP on 15/5/3.
//  Copyright (c) 2015年 LSP. All rights reserved.
//

#import "UIButton+Loading.h"

@implementation UIButton (Loading)

-(void)startLoading{
    UIActivityIndicatorView *aiv = [[UIActivityIndicatorView alloc]initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
    
    aiv.center = self.center;
    
    aiv.tag = 97561;
    [aiv startAnimating];
    
    self.enabled = NO;
    
    [self.superview addSubview:aiv];
}

-(void)endLoading{
    self.enabled = YES;
    
    UIActivityIndicatorView *aiv = (UIActivityIndicatorView *)[self.superview viewWithTag:97561];
    [aiv stopAnimating];
    
    [aiv removeFromSuperview];
}

@end

File: Sources/UIButton/UIButton+Loading.swift
//
//  UIButton+Loading.swift
//  SPUIKitCategory module
//
//  Created by LSP on 2020/12/22.
//  Copyright © 2020 LSP. All rights reserved.
//

#if !os(macOS)
import UIKit

extension UIButton {
    
    ///开始加载
    public
    func startLoading() {
        let aiv = UIActivityIndicatorView(style: .gray)
        aiv.center = self.center
        aiv.tag = 93546569
        aiv.startAnimating()
        self.isEnabled = false
        self.superview?.addSubview(aiv)
    }
    
    ///结束加载
    public
    func endLoading() {
        self.isEnabled = true
        
        if let aiv = self.superview?.viewWithTag(93546569) as? UIActivityIndicatorView {
            aiv.stopAnimating()
            aiv.removeFromSuperview()
        }
    }
}
#endif
